homeassistant/components/webhook/__init__.py

```diff
@@ -87,10 +87,17 @@ def async_generate_id() -> str:
 
 @callback
 @bind_hass
-def async_generate_url(hass: HomeAssistant, webhook_id: str) -> str:
+def async_generate_url(
+    hass: HomeAssistant,
+    webhook_id: str,
+    allow_internal: bool = True,
+    allow_external: bool = True,
+    allow_ip: bool | None = None,
+    prefer_external: bool | None = True,
+) -> str:
     """Generate the full URL for a webhook_id."""
     return (
-        f"{get_url(hass, prefer_external=True, allow_cloud=False)}"
+        f"{get_url(hass,allow_internal=allow_internal, allow_external=allow_external, allow_cloud=False, allow_ip=allow_ip, prefer_external=prefer_external,)}"
```

Comment:
Please break long strings around max 88 characters per line.

Reply:
Side note: I'm surprised that ruff accepts missing whitespace after a comma.

Reply:
https://github.com/home-assistant/core/pull/118910 Ruff/black didn't really care at all about that line, maybe cause it's a multi-line `f`-string